Add photo taken to profile during profile creation

Send to server with other profile data

Getting a crash after pushing the Start Binding button....will fix

```
11-16 13:36:59.949 1630-1630/ca.binder E/AndroidRuntime: FATAL EXCEPTION: main
                                                         Process: ca.binder, PID: 1630
                                                         java.lang.IllegalStateException: Could not execute method of the activity
                                                             at android.view.View$1.onClick(View.java:3823)
                                                             at android.view.View.performClick(View.java:4438)
                                                             at android.view.View$PerformClick.run(View.java:18422)
                                                             at android.os.Handler.handleCallback(Handler.java:733)
                                                             at android.os.Handler.dispatchMessage(Handler.java:95)
                                                             at android.os.Looper.loop(Looper.java:136)
                                                             at android.app.ActivityThread.main(ActivityThread.java:5001)
                                                             at java.lang.reflect.Method.invokeNative(Native Method)
                                                             at java.lang.reflect.Method.invoke(Method.java:515)
                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
                                                             at dalvik.system.NativeStart.main(Native Method)
                                                          Caused by: java.lang.reflect.InvocationTargetException
                                                             at java.lang.reflect.Method.invokeNative(Native Method)
                                                             at java.lang.reflect.Method.invoke(Method.java:515)
                                                             at android.view.View$1.onClick(View.java:3818)
                                                             at android.view.View.performClick(View.java:4438) 
                                                             at android.view.View$PerformClick.run(View.java:18422) 
                                                             at android.os.Handler.handleCallback(Handler.java:733) 
                                                             at android.os.Handler.dispatchMessage(Handler.java:95) 
                                                             at android.os.Looper.loop(Looper.java:136) 
                                                             at android.app.ActivityThread.main(ActivityThread.java:5001) 
                                                             at java.lang.reflect.Method.invokeNative(Native Method) 
                                                             at java.lang.reflect.Method.invoke(Method.java:515) 
                                                             at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785) 
                                                             at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601) 
                                                             at dalvik.system.NativeStart.main(Native Method) 
                                                          Caused by: java.lang.ClassCastException: android.widget.Spinner cannot be cast to android.widget.EditText
                                                             at ca.binder.FirstLaunchActivity.text(FirstLaunchActivity.java:180)
                                                             at ca.binder.FirstLaunchActivity.createProfile(FirstLaunchActivity.java:135)
                                                             at ca.binder.FirstLaunchActivity.onFinishSetup(FirstLaunchActivity.java:99)
```

--- a/src/main/java/ca/binder/FirstLaunchActivity.java
+++ b/src/main/java/ca/binder/FirstLaunchActivity.java
@@ -1,25 +1,25 @@
 package ca.binder;
 
 import android.app.Activity;
-import android.app.ProgressDialog;
 import android.content.Intent;
 import android.graphics.Bitmap;
-import android.os.AsyncTask;
 import android.os.Bundle;
 import android.provider.MediaStore;
 import android.util.Log;
 import android.view.View;
-import android.widget.*;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Spinner;
+import android.widget.Toast;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-import ca.binder.android.DeviceInfo;
 import ca.binder.android.InternalPhoto;
 import ca.binder.domain.Course;
-import ca.binder.domain.Profile;
 import ca.binder.domain.ProfileBuilder;
 import ca.binder.remote.Callback;
 import ca.binder.remote.Server;
@@ -156,7 +156,7 @@ public class FirstLaunchActivity extends Activity {
                     onProfileCreateFailure();
                 }
             }
-        });
+        }).run();
     }
 
     /**
@@ -177,6 +177,9 @@ public class FirstLaunchActivity extends Activity {
     }
 
     private String text(int inputId) {
+        if (inputId == R.id.year_input) {
+            return ((Spinner) findViewById(inputId)).getSelectedItem().toString();
+        }
         return ((EditText)(findViewById(inputId))).getText().toString();
     }
 

--- a/src/main/java/ca/binder/SuggestionViewActivity.java
+++ b/src/main/java/ca/binder/SuggestionViewActivity.java
@@ -22,7 +22,9 @@ import butterknife.ButterKnife;
 import ca.binder.android.DeviceInfo;
 import ca.binder.domain.Suggestion;
 import ca.binder.domain.SuggestionReaction;
+import ca.binder.remote.Callback;
 import ca.binder.remote.Server;
+import ca.binder.remote.request.AsyncServerRequest;
 import ca.binder.remote.request.DislikeSuggestionRequest;
 import ca.binder.remote.request.GetSuggestionsRequest;
 import ca.binder.remote.request.LikeSuggestionRequest;
@@ -31,6 +33,7 @@ import ca.binder.remote.request.LikeSuggestionRequest;
 public class SuggestionViewActivity extends Activity {
 
 	private final String LOG_TAG = "SwipeViewAcitivity";
+	private ProgressDialog dialog;
 
 	@Bind(R.id.swipe_view_card_container)
 	SwipeFlingAdapterView flingContainer;
@@ -46,15 +49,29 @@ public class SuggestionViewActivity extends Activity {
 		ButterKnife.bind(this);
 
 		viewSwitcher = (ViewSwitcher) findViewById(R.id.suggestionViewSwitcher);
-
+		dialog = new ProgressDialog(SuggestionViewActivity.this);
 		suggestionsToShow = new ArrayList<>();
 
 
 		//createTestSuggestions();
 
 		// get suggestions from api asynchronously
-		GetSuggestionsTask task = new GetSuggestionsTask();
-		task.execute(this);
+//		GetSuggestionsTask task = new GetSuggestionsTask();
+//		task.execute(this);
+		Server server = Server.standard(this);
+		new AsyncServerRequest<>(server, new GetSuggestionsRequest(), new Callback<List<Suggestion>>() {
+			@Override
+			public void use(List<Suggestion> suggestions) {
+				//Executed after request finishes
+				if (suggestions != null) {
+					onGetSuggestionsSuccess(suggestions);
+				} else {
+					onGetSuggestionsFailure();
+				}
+			}
+		});
+		this.dialog.setMessage("Loading...");
+		this.dialog.show();
 
 
 		/**
@@ -114,6 +131,26 @@ public class SuggestionViewActivity extends Activity {
 			}
 		});
 
+	}
+
+	private void onGetSuggestionsFailure() {
+		// TODO
+		Log.e("GetSuggestions", "Failed");
+		dialog.dismiss();
+	}
+
+	private void onGetSuggestionsSuccess(List<Suggestion> suggestions) {
+		// TODO
+		suggestionsToShow = suggestions;
+		Log.d("GetSuggestions", "Done!");
+		Log.d("GetSuggestions", "# of suggestions: " + suggestionsToShow.size());
+
+
+		adapter = new SuggestionAdapter(SuggestionViewActivity.this, R.layout.suggestion_card_view, suggestionsToShow);
+		flingContainer.setAdapter(adapter);
+		adapter.notifyDataSetChanged();
+		checkForEmptySuggestionList(adapter.getCount());
+		dialog.dismiss();
 	}
 
 	/**

